### ISSUE TITLE
Using GHA cache always leads to `Image` update

### ISSUE THREAD

**Issue by tlinhart** [pulumi/pulumi-docker-build#75]
### What happened?

We use GitHub Actions for CI/CD so I would like to use GHA cache for the Docker image build process. However the `Image` resource always gets updated because the `cacheFrom` and `cacheTo` differ. As per the [docs](https://www.pulumi.com/registry/packages/docker-build/api-docs/image/#cachefromgithubactions), when `url` and `token` arguments are not provided for the GHA `cacheFrom` and `cacheTo`, they are taken from the environment variables `ACTIONS_RUNTIME_URL` and `ACTIONS_RUNTIME_TOKEN`. However, even though the runtime URL stays the same, the token is always different as it is a JWT token related to the current workflow run and job (see e.g. [here](https://github.com/tonistiigi/go-actions-cache/blob/master/api.md#authentication)). I expect Pulumi not to trigger an update when `url` and `token` are not provided explicitely.

### Example

This is the relevant part of the Pulumi program:

```
image_name = pulumi.Output.concat(repository.repository_url, ":latest")
if os.getenv("GITHUB_ACTIONS"):
    cache_from = docker_build.CacheFromArgs(
        gha=docker_build.CacheFromGitHubActionsArgs()
    )
    cache_to = docker_build.CacheToArgs(
        gha=docker_build.CacheToGitHubActionsArgs(
            mode=docker_build.CacheMode.MAX, ignore_error=True
        )
    )
else:
    cache_from = docker_build.CacheFromArgs(
        registry=docker_build.CacheFromRegistryArgs(ref=image_name)
    )
    cache_to = docker_build.CacheToArgs(
        inline=docker_build.CacheToInlineArgs()
    )
image = docker_build.Image(
    "xxx",
    context=docker_build.BuildContextArgs(location=".."),
    platforms=[docker_build.Platform.LINUX_AMD64],
    tags=[image_name],
    cache_from=[cache_from],
    cache_to=[cache_to],
    push=True,
    registries=[
        docker_build.RegistryArgs(
            address=repository.repository_url,
            username=auth_token.user_name,
            password=pulumi.Output.secret(auth_token.password),
        )
    ],
)
```

This is the relevant part of the GitHub Actions log from the `pulumi up` step:

```
    ~ docker-build:index:Image: (update)
          [id=sha256:xxx]
          [urn=urn:pulumi:xxx-test::xxx::docker-build:index:Image::xxx]
          [provider=urn:pulumi:xxx-test::xxx::pulumi:providers:docker-build::default_0_0_2::1c6deb57-2a3f-4817-8d97-752bb9f399d4]
        ~ cacheFrom  : [
            ~ [0]: {
                      disabled: false
                    ~ gha     : {
                        ~ token: "***" => ***
                          url  : "https://pipelinesghubeus3.actions.githubusercontent.com/xxx"
                      }
                  }
          ]
        ~ cacheTo    : [
            ~ [0]: {
                      disabled: false
                    ~ gha     : {
                          ignoreError: true
                          mode       : "max"
                        ~ token      : "***" => ***
                          url        : "https://pipelinesghubeus3.actions.githubusercontent.com/xxx"
                      }
                  }
          ]
```

### Output of `pulumi about`

```
CLI          
Version      3.116.1
Go Version   go1.22.2
Go Compiler  gc

Host     
OS       ubuntu
Version  24.04
Arch     x86_64

Backend        
Name           xxx
URL            s3://xxx
User           xxx
Organizations  
Token type     personal

Pulumi locates its logs in /tmp by default
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by blampe** [comment]
Ah, great catch. We take ignore changes to the registry's password for the same reason, we should do something similar for the GHA token.

**Comment by pose** [comment]
@blampe why we cannot just simply ignore the env variable for the token and call it a day? What's the customer experience we are trying to support by caching that token?

**Comment by blampe** [comment]
> [@blampe](https://github.com/blampe) why we cannot just simply ignore the env variable for the token and call it a day? What's the customer experience we are trying to support by caching that token?

That's true, we might be able to remove these inputs https://github.com/pulumi/pulumi-docker-build/blob/40c723d82f2d9909944ae8c06eee69a6f5777f46/provider/internal/cache.go#L158-L159. If a user needs to override these they would still be able to do so with the env var. We'll probably need to update some tests, but that's not a big deal. This isn't 1.0 yet so a breaking change is fine.

I think I included these as inputs since it's historically been the convention to store anything that contributes to the program in state, but issues like this always come up.

**Comment by nathan-knight** [comment]
> Hi [@pose](https://github.com/pose), thanks. I've just encountered this in my GHA workflow:
> 
> ```
> error: 2 failures decoding:
>   	cacheFrom[0]: An error occurred decoding 'ImageState.cacheFrom[0]': 1 failures decoding:
>   	gha: An error occurred decoding 'CacheFrom.gha': 2 failures decoding:
>   	token: Unrecognized field 'token' on 'internal.CacheFromGitHubActions'
>   	url: Unrecognized field 'url' on 'internal.CacheFromGitHubActions'
>   	cacheTo[0]: An error occurred decoding 'ImageState.cacheTo[0]': 1 failures decoding:
>   	gha: An error occurred decoding 'CacheTo.gha': 2 failures decoding:
>   	token: Unrecognized field 'token' on 'internal.CacheToGitHubActions'
>   	url: Unrecognized field 'url' on 'internal.CacheToGitHubActions'
> ```
> 
> I guess your suggestions would not help with this error, would they?

What I did to resolve this was:
1. Go back to the version I was on
2. Remove the GHA cache configuration and deploy
3. Update to the latest version
4. Bring the GHA cache configuration back and deploy again
